modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/AuditBarChart.js

```diff
@@ -81,13 +99,76 @@ export default function AuditBarChart({rtl, vocabularies}) {
 		? [BAR_CHART.dotRadiusMin]
 		: [BAR_CHART.legendHeight + BAR_CHART.dotRadiusMin];
 
+	const [checkboxes, setCheckbox] = useState(legendCheckboxes);
+
+	useEffect(() => {
+		if (Object.keys(colors).length) {
+			const head =
+				document.head || document.getElementsByTagName('head')[0];
+			const style = document.createElement('style');
+			style.id = 'legend-checkboxes';
+			style.type = 'text/css';
+
+			{
+				Object.entries(colors).forEach(([dataKey, color]) => {
+					style.textContent += `.custom-control-color-${dataKey}.custom-control-input:checked ~ 
+					.custom-control-label::before {
+						background-color: ${color};
+						border-color: ${color};
+				}
+				.custom-control-color-${dataKey}.custom-control-input:not(:checked) ~ 
+					.custom-control-label::before {
+						border-color: ${color};
+				}`;
+				});
+			}
+
+			head.appendChild(style);
+		}
+
+		return () => {
+			if (Object.keys(colors).length) {
+				const head =
+					document.head || document.getElementsByTagName('head')[0];
+				const style = document.getElementById('legend-checkboxes');
+				head.removeChild(style);
+			}
+		};
+	}, [colors]);
+
+	const renderLegend = (props) => {
+		const {payload} = props;
+
+		return (
+			<>
```

Comment:
Is this fragment necessary? I think you can return the `.map` without having to wrap it.

```jsx
() => {
    return payload.map(entry => ...)
}
```

Reply:
That's right! Thanks!